### PR TITLE
More io.Data uses.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: lib/crypto/aes.toit [hunk 1]
```diff
@@ -2,6 +2,8 @@
 // Use of this source code is governed by an MIT-style license that can be
 // found in the lib/LICENSE file.
 
+import ..io as io
```
Comment (kasperl) [inline lib/crypto/aes.toit:5]: Shouldn't this just be `import io` or is this just easier to work with for now?

File: lib/net/modules/tcp.toit [hunk 2]
```diff
@@ -197,7 +197,13 @@ tcp-listen_ socket-resource-group address port backlog:
   #primitive.tcp.listen
 
 tcp-write_ socket-resource-group descriptor data from to:
-  #primitive.tcp.write
+  #primitive.tcp.write: | error |
+    if error != "WRONG_BYTES_TYPE": throw error
```
Comment (kasperl) [inline lib/net/modules/tcp.toit:201]: Add a comment to explain why this isn't using `io.primitive-redo-chunked-io-data_`.

File: tests/socket-test.toit [hunk 3]
```diff
@@ -85,5 +111,5 @@ sleepy-reader port:
         expected = "Message for sleepy reader $index"
       expect str.size < expected.size
   print "End at $index"
-  expect-equals index 100
+  expect-equals index iterations
```
Comment (kasperl) [inline tests/socket-test.toit:114]:
```suggestion
  expect-equals iterations index
```

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: lib/net/modules/tcp.toit (hunk 2 above)
Comment (floitsch) [reply lib/net/modules/tcp.toit:201]: done.

File: tests/socket-test.toit (hunk 3 above)
Comment (floitsch) [reply tests/socket-test.toit:114]: done.

**Pull request merged**